tools/sctrace/README.md

```diff
@@ -0,0 +1,357 @@
+# Syscall-Compliance-Trace
+
+Syscall-Compliance-Trace (sctrace) is a powerful system call compliance
+verification tool that analyzes and validates system call against user-defined
+compliance patterns. Written in SCML (System Call Matching Language), these
+patterns describe supported functionality of system calls.
+sctrace supports both real-time monitoring of running programs and post-analysis of
+existing trace logs, providing comprehensive insights into system call compliance
+with intuitive pattern matching and visual feedback.
+
+## Features
+
+- **Pattern-based filtering**: Define system call patterns using SCML syntax
+- **Dual mode operation**:
+  - Online mode: Real-time tracing of running programs
+  - Offline mode: Analysis of existing strace log files
+- **Quiet mode**: Minimal output showing only unsupported calls
+- **Multi-threaded support**: Automatic handling of multi-threaded program traces with syscall reconstruction
+
+## Installation
+
+Make sure you have Rust installed, then build the project:
+
+```bash
+cargo build --release
+```
+
+The binary will be available at `target/release/sctrace`.
+
+## Prerequisites
+
+- **strace** version 5.15 or higher (for online mode)
+- Rust toolchain
+
+## Usage
+
+### Basic Syntax
+
+```bash
+sctrace <scml_file> [OPTIONS] -- [program] [args...]
+```
+
+### Online Mode (Real-time tracing)
+
+Trace a program in real-time:
+
+```bash
+sctrace patterns.scml -- ls -la
+sctrace patterns.scml --quiet -- ./my_program arg1 arg2
+```
+
+### Offline Mode (Log file analysis)
+
+Analyze an existing strace log file:
+
+```bash
+sctrace patterns.scml --input trace.log
```

Comment:
My understanding is that we need a specific format of trace log. So the user documentation needs to describe the required strace options.